src/fowt_ml/linear_models.py

```diff
@@ -0,0 +1,100 @@
+"""Module to handle linear models."""
+
+from dataclasses import dataclass
+import sklearn.linear_model as lm
+import sklearn.metrics as sm
+from numpy.typing import ArrayLike
+
+
+@dataclass
+class Estimator:
+    name: str
+    func: callable
+    reference: str = "sklearn"
+
+
+@dataclass
+class Metric:
+    name: str
+    func: callable
+    reference: str = "sklearn"
+
+
+class LinearModels:
+    """Class to handle linear models and metrics for comparison."""
+
+    ESTIMATOR_NAMES = {
+        "LinearRegression": lm.LinearRegression(),
+        "RidgeRegression": lm.Ridge(),
+        "LassoRegression": lm.Lasso(),
+        "ElasticNetRegression": lm.ElasticNet(),
+        "LeastAngleRegression": lm.Lars(),
+    }
+    METRICS_NAMES = {
+        "root_mean_squared_error": sm.root_mean_squared_error,
+        "mean_squared_error": sm.mean_squared_error,
+        "r2_score": sm.r2_score,
+        "mean_absolute_error": sm.mean_absolute_error,
+    }
+
+    def __init__(self, estimator: str | Estimator, kwargs: dict = None) -> None:
+        """Initialize the class with the estimator."""
+        if isinstance(estimator, str):
+            if not self.ESTIMATOR_NAMES.get(estimator):
+                msg = (
+                    f"estimator {estimator} not supported. "
+                    f"Choose one of {list(self.ESTIMATOR_NAMES.keys())}"
+                    f"or pass a Estimator instance."
+                )
+                raise ValueError(msg)
+            self.estimator = Estimator(estimator, self.ESTIMATOR_NAMES.get(estimator))
+        elif isinstance(estimator, Estimator):
+            self.estimator = estimator
+            # TODO: validate if model function is a callable and valid
+        else:
+            raise ValueError("model must be a string or a Estimator instance.")
+
+        # fill the estimator with the kwargs
+        if kwargs:
+            for key, value in kwargs.items():
+                setattr(self.estimator.func, key, value)
```

Comment:
Estimator objects in scikit-learn should have a `set_params` method (see [link](https://scikit-learn.org/dev/developers/develop.html#estimators)), so maybe one could do:
```suggestion
        self.estimator.func.set_params(**kwargs)
```

Reply:
cool, I didn't know the method. thanks for the suggestion, it is fixed now.